tools/toolchains/__init__.py

```diff
@@ -702,6 +719,7 @@ def get_arch_file(self, objects):
 
     # THIS METHOD IS BEING CALLED BY THE MBED ONLINE BUILD SYSTEM
     # ANY CHANGE OF PARAMETERS OR RETURN VALUES WILL BREAK COMPATIBILITY
+    @check_toolchain_path
```

Comment:
Then it would have to take in a parameter to check the existence of. At the Class construction time (when this gets called) we don't know what the directory we need to check should be, so we can't pass it.